tech_questions.py

```diff
@@ -0,0 +1,187 @@
+'''Question 1: Design a function in Python for creating a
+ lookup table from a collection of objects.  Your function
+  will take a list of objects (of a particular type) and 
+  the name of a property to use as the primary key for the 
+  lookup.  The result should be a hash table that can be used 
+  to lookup objects by values of that property.  The lookup 
+  table should only store references to the objects and avoid 
+  copying the source data.
+'''
+
+
+
+'''
+ASSUMPTIONS: 
+	- original objects' values will not change, and thus
+	need to be reflected in the hash table?
+	- objects may have any unknown amount/type of
+arguements? "Of a particular type" type doesn't really
+describe what assumptions I should have. *args **kwargs ?
+
+
+Confusion...you specify using 'a property to use as the primary key'
+but then go on to say that this should be a hash table.
+
+
+What is the hash to be made out of? Random? Combination
+of object's attributes?
+
+
+Need to not copy, need to be reference
+'''
+
+
+# pretty print
+from pprint import pprint
+import hashlib
+
+
+
+
+
+class particular_type_of_object:
+	def __init__(self, hostname, ip, os, backup_enabled, hash_value):
+		'''hash_value = which value should determine the hash, 'ip', 'hostname',
+		'etc' '''
+		self.hostname = hostname
+		self.ip = ip
+		self.os = os
+		self.backup_enabled = backup_enabled
+		# they never specified what the hash was to be made from
+		self.values = {'ip':self.ip,
+						'hostname':self.hostname,
+						'os':self.os,
+						'backup_enabled':self.backup_enabled}
```

Comment:
I don't think this `values` property makes much sense. These key=>values are identical data to the build-in `__dict__` property, so you can probably delete this.